### ISSUE TITLE
Low accuracy when the backdoor model is set to eval mode

### ISSUE THREAD

**Issue by zeabin** [sunbelbd/invisible_backdoor_attacks#1]
Hi, @khoadoan @sunbelbd I run the pytorch code using the default command
```
python pytorch/lira_trigger_generation.py --dataset cifar10 --clsmodel vgg11 --path experiments/ --epochs 50 --train-epoch 1 --mode all2one --target_label 0 --epochs_per_external_eval 10 --cls_test_epochs 5 --verbose 2 --batch-size 128 --alpha 0.5 --eps 0.01 --avoid_clsmodel_reinitialization
python pytorch/lira_backdoor_injection.py --dataset cifar10 --clsmodel vgg11 --path experiments/ --epochs 50 --train-epoch 1 --mode all2one --target_label 0 --epochs_per_external_eval 10 --cls_test_epochs 5 --verbose 2 --batch-size 128 --alpha 0.5 --eps 0.01 --avoid_clsmodel_reinitialization --test_eps 0.01 --test_alpha 0.5 --test_epochs 250 --test_lr 0.01 --schedulerC_lambda 0.1 --schedulerC_milestones 50,100,150,200 --test_use_train_best
```
to train a vgg11 model, and find that the performance is (clean 0.8695 poison 0.9939) in train mode, but drop to (clean 0.3894 poison 0.7380) when the model is set to eval mode.

The same performance drop happens in PreActResNet18, from (0.9209, 0.9992) to (0.5737, 0.5001)

I notice that in `lira_backdoor_injection.py`, the performance in test dataset is tested without setting the model to eval mode. Is that a problem or did I make something wrong? 

Can you share the pretrained backdoor model?

**Comment by Fulib** [comment]
do you find a solution for that,i have the same problem about this
